app/controllers/ChoiceController.scala

```diff
@@ -60,15 +61,31 @@ class ChoiceController @Inject()(
   }
 
   private def proceed(choice: Choice)(implicit request: AuthenticatedRequest[AnyContent]): Future[Result] = choice match {
-    case Choice.Arrival              => saveAndRedirect(ArrivalAnswers(), movements.routes.ConsignmentReferencesController.displayPage())
-    case Choice.RetrospectiveArrival => saveAndRedirect(RetrospectiveArrivalAnswers(), movements.routes.ConsignmentReferencesController.displayPage())
-    case Choice.Departure            => saveAndRedirect(DepartureAnswers(), movements.routes.ConsignmentReferencesController.displayPage())
-    case Choice.AssociateUCR         => saveAndRedirect(AssociateUcrAnswers(), consolidations.routes.MucrOptionsController.displayPage())
-    case Choice.DisassociateUCR      => saveAndRedirect(DisassociateUcrAnswers(), consolidations.routes.DisassociateUCRController.display())
-    case Choice.ShutMUCR             => saveAndRedirect(ShutMucrAnswers(), consolidations.routes.ShutMucrController.displayPage())
-    case Choice.ViewSubmissions      => Future.successful(Redirect(routes.ViewSubmissionsController.displayPage()))
+    case Choice.Arrival =>
+      saveAndRedirect(ArrivalAnswers.fromQueryUcr, movements.routes.ConsignmentReferencesController.displayPage())
+    case Choice.RetrospectiveArrival =>
+      saveAndRedirect(RetrospectiveArrivalAnswers.fromQueryUcr, movements.routes.ConsignmentReferencesController.displayPage())
+    case Choice.Departure =>
+      saveAndRedirect(DepartureAnswers.fromQueryUcr, movements.routes.ConsignmentReferencesController.displayPage())
```

Comment:
Nit: I like the idea with using function parameter. To me it would be more natural to add `fromQueryUcr` method to `Answers` trait and then just call it in `saveAndRedirect`. The signature of this method would stay the same in that case.
But I am just more used to OOP solutions.